src/librustc_mir/interpret/memory.rs

```diff
@@ -290,6 +290,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
 
         // Let the machine take some extra action
         let size = alloc.size;
+        M::notify_deallocation(&self.extra, &alloc, ptr);
```

Comment:
Please move this to before `alloc_map.remove`, so that it truly is before deallocation (like our hooks usually are).